app/src/main/java/com/example/notes2text/adapters/FileListAdaptor.java

```diff
@@ -3,51 +3,78 @@
 import android.content.Context;
 import android.content.Intent;
 import android.view.LayoutInflater;
-import android.view.MenuItem;
 import android.view.View;
 import android.view.ViewGroup;
-import android.widget.ImageView;
-import android.widget.PopupMenu;
-import android.widget.TextView;
 import android.widget.Toast;
 
 import androidx.annotation.NonNull;
 import androidx.recyclerview.widget.RecyclerView;
 
 import com.example.notes2text.R;
-import com.example.notes2text.adapters.fragments.DirectoryAccessController;
+import com.example.notes2text.entities.FileViewHolder;
+import com.example.notes2text.usecases.DirectoryUseCases.FileViewHolderFactory;
+import com.example.notes2text.usecases.DirectoryUseCases.ViewHolderAbsFactory;
 import com.example.notes2text.usecases.FileMenuFactory;
-import com.example.notes2text.usecases.FileMenuInputBoundary;
-import com.example.notes2text.usecases.FileMenuInteractor;
 import com.example.notes2text.usecases.FileOpenInteractor;
 
 import java.io.File;
 
-public class FileListAdaptor extends RecyclerView.Adapter<FileListAdaptor.ViewHolder> {
+/**
+ * A recycler view adapter to load lists of files in directory format.
+ */
+public class FileListAdaptor extends RecyclerView.Adapter<FileViewHolder> {
+    //Adjusted to use FileViewHolder rather than inner class viewholder.
 
     Context context;
     protected File[] fileList;
 
+    protected ViewHolderAbsFactory viewHolderCreator = new FileViewHolderFactory();
 
+
+    //TODO: Create interface to segregate this from the file opener.
     FileOpenInteractor fileOpener = new FileOpenInteractor();
 
+    /**
+     * Creates a recyclerview.adapter for file display.
+     *
+     * @param context The application context. Required to interact with Android screen elements.
+     * @param fileList The list of files to display in the recycler view.
+     */
     public FileListAdaptor(Context context, File[] fileList){
         super();
         this.context = context;
         this.fileList = fileList;
     }
 
     //A function that creates the ViewHolder required for the recyclerview for file list.
+
+    /**
+     *
+     * @param source The ViewGroup into which the new View will be added after it is bound to
+     *               an adapter position.
+     * @param ViewType The view type of the new View.
+     *
+     * @return FileViewHolder, the file holder.
+     */
+    @NonNull
     @Override
-    public ViewHolder onCreateViewHolder(ViewGroup source, int ViewType){
+    public FileViewHolder onCreateViewHolder(@NonNull ViewGroup source, int ViewType){
 
         View view = LayoutInflater.from(context).inflate(R.layout.file_holder_view_model, source, false);
-        return new ViewHolder(view);
+//        return new FileViewHolder(view);
```

Comment:
Make sure to remove this line if its no longer needed.

Reply:
Fixed with commit.